pkgs/applications/editors/bluej/default.nix

```diff
@@ -11,26 +12,52 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-KhNhJ2xsw1g2yemwP6NQmJvk4cxZAQQNPEUBuLso5qM=";
   };
 
-  nativeBuildInputs = [ makeWrapper ];
+  nativeBuildInputs = [ wrapGAppsHook ];
+  buildInputs = [ glib ];
 
-  unpackPhase = ''
-    ar xf $src
-    tar xf data.tar.xz
+  sourceRoot = ".";
+  preUnpack = ''
+    unpackCmdHooks+=(_tryDebData)
+    _tryDebData() {
+      if ! [[ "$1" =~ \.deb$ ]]; then return 1; fi
+      ar xf "$1"
+      if ! [[ -e data.tar.xz ]]; then return 1; fi
+      unpackFile data.tar.xz
+    }
   '';
 
   installPhase = ''
-    mkdir -p $out
-    cp -r usr/* $out
+    runHook preInstall
+
+    if [ -n "$prefix" ]; then
+        mkdir -p "$prefix"
+    fi
+
+    mkdir -p "$out"
+
+    if shopt -q dotglob; then dotglobOpt=$?; else dotglobOpt=$?; fi
+    shopt -s dotglob
+    for file in usr/*; do
+      cp -R "$file" "$out"
+    done
+    if (( !dotglobOpt )); then shopt -u dotglob; fi
```

Comment:
What's your reasoning for making this more complex? I don't see why doing things this way would be better.

Reply:
That option can't be something different than the default value. Thats why we have the build sandbox.